physics/GFS_MP_generic_post.F90

```diff
@@ -111,6 +123,68 @@ subroutine GFS_MP_generic_post_run(
         rain(i) = rainc(i) + frain * rain1(i) ! time-step convective plus explicit
       enddo
 
+! compute surface snowfall, graupel/sleet, freezing rain and precip ice density
+      if (imp_physics == imp_physics_gfdl .or. imp_physics == imp_physics_thompson .or. imp_physics == imp_physics_nssl ) then
+         do i = 1, im
+            if (gt0(i,1) .le. 273) then
+               frzr(i) = frzr(i) + rain0(i)
+               frzrb(i) = frzrb(i) + rain0(i)
+            endif
+            tsnowp(i)  = tsnowp(i)  + snow0(i)
+            tsnowpb(i)  = tsnowpb(i)  + snow0(i)
+            frozr(i) = frozr(i) + graupel0(i)
+            frozrb(i) = frozrb(i) + graupel0(i)
+         enddo
+!Compute the variable precip ice density for specific LSM schemes and options
+         if ( lsm == lsm_ruc .or. lsm == lsm_noahmp .and. iopt_snf == 5 .or. vrbliceden_noah == .true.) then
```

Comment:
Shouldn't `lsm == lsm_noahmp .and. iopt_snf == 5` be in parentheses? Maybe not technically since AND has precedence over OR and are evaluated first, but at least for code clarity?